arch/arm/mach-sc5xx/soc.c

```diff
@@ -45,23 +46,89 @@ int arch_cpu_init(void)
 	return 0;
 }
 
+
+
+
+void print_device_type(u32 device_type)
+{
+	printf("[Detected] Device: "); 
+	if(device_type == GRFN_SC58X_JTAGID)
+		printf("sc58x\n");
+	else if(device_type == GRFN_SC59X_JTAGID)
+		printf("sc59x\n");
+	else
+		printf("unknown [%04x]\n",device_type);
+}
+
+/* 
+* Revision ID is similarly coded for different boards
+* and as a result, is currently set to be shared by different 
+* devices 
+* */
+void print_si_version(u32 si_version)
+{
+	printf("[Detected] Silicon version: ");
+	if(si_version == GRFN_SC58X_SI_REVID_0_0)
```

Comment:
this is absolutely optional, but since my MISRA days I've stuck to using `if ($VALUE == variable)` and not the other way around, as it would give a compilation error if someone dropped an `=`, instead of turning into an assignment statement :) just a safe practice.